cognite/powerops/case/case.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import logging
+import tempfile
+from typing import Any
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+class Case:
+    r"""
+    Wrapper around YAML file for SHOP, describing a case.
+
+    Main features:
+     * Values can be retrieved or set / changed like with a dict.
+        * Dot syntax is also supported.
+     * Can hold references to additional files ("extra files").
+        * These are just paths to files, their content is not accessed here.
+     * Loading from string or file (with `Case.load_yaml(path_to_yaml)`).
+     * Saving to yaml (with `case.save_yaml(path_to_file)`).
+
+    Examples:
+       >>> case = Case('''
+       ... foo:
+       ...   bar1: 11
+       ...   bar2: 22
+       ... ''')
+       >>> case["foo"]["bar2"] = 202
+       >>> case.yaml
+       'foo:\n  bar1: 11\n  bar2: 202\n'
+       >>> case["foo.bar1"] = 101
```

Comment:
Are the numbers suppose to be 11 and 22? Not 101 and 202?

Reply:
Example was meant to show that you can edit the values. INitially they were `11` and `22`, then got updated to `101` and `202`.

Reply:
Split the examples up. I thought splitting like this would confuse doctests, but it runs just fine 🥳